aws-s3-private-bucket/variables.tf

```diff
@@ -34,3 +34,26 @@ variable "abort_incomplete_multipart_upload_days" {
   description = "Number of days after which an incomplete multipart upload is canceled."
   default     = 14
 }
+
+variable "lifecycle_rule" {
+  description = "List of maps containing configuration of object lifecycle management."
+  type        = any
```

Comment:
Interesting– it had not occurred to me that we can't use structural typing here, since we want to allow values to be be absent - https://www.terraform.io/docs/configuration/types.html#structural-types

Reply:
After reading those docs, it's not clear to me why we couldn't do this? Could it be typed as `list(object(...))` and then have the default be an empty list?

Reply:
If we use `object(...)` we'd have to define the named keys, which might be absent right? If we have the default be an empty list, how would we pass in the existing rule?